core/crypto/sm3.c

```diff
@@ -84,6 +84,7 @@ static void sm3_process(struct sm3_context *ctx, const uint8_t data[64])
 
 #define SHL(x, n)	((x) << (n))
 #define ROTL(x, n)	(SHL((x), (n)) | ((x) >> (32 - (n))))
+#define ROTL2(x, n)	(((n) <= 32) ? ROTL((x), (n)) : ROTL((x), (n) - 32))
```

Comment:
This will probably emit conditional branch, which is not good in hot path.

Why about `ROTL((x), (n) & 0x1F )` ?

Reply:
Or even better, update ROTL to always mask the `n` parameter like that.

Reply:
Agreed, much better.